src/metaprob/state.clj

```diff
@@ -58,7 +57,7 @@
                     (vector? state) {:value (nth state key)}
                     (map? state) (get state key)
                     true (assert false ["not a state" state])))]
-    (assert (not (= val nil))
+    (assert (some? val)
```

Comment:
Very hackish, not comfortable with it - I would have to look up the definitions of `some?` to see how it works (and the documentation probably wouldn't tell me, and I would go nonlinear again)